libcnb-test/src/test_runner.rs

```diff
@@ -0,0 +1,216 @@
+/// # }
+/// TestRunner::default().run_test(
+///     TestConfig::new("heroku/builder:22", "test-fixtures/app").buildpacks(vec![
+///         BuildpackReference::Other(String::from("heroku/openjdk")),
+///         BuildpackReference::Crate,
+///     ]),
+///     |context| {
+///         assert_contains!(context.pack_stdout, "---> Maven Buildpack");
+///         assert_contains!(context.pack_stdout, "---> Installing Maven");
+///         assert_contains!(context.pack_stdout, "---> Running mvn package");
+///
+///         context
+///             .prepare_container()
+///             .expose_port(12345)
+///             .start_with_default_process(|container| {
+///                 assert_eq!(
+///                     call_test_fixture_service(
+///                         container.address_for_port(12345).unwrap(),
+///                         "Hagbard Celine"
+///                     )
+///                     .unwrap(),
+///                     "enileC drabgaH"
+///                 );
+///             });
+///     },
+/// );
+/// ```
+pub struct TestRunner {
+    pub(crate) docker: Docker,
+    pub(crate) tokio_runtime: tokio::runtime::Runtime,
+}
+
+impl Default for TestRunner {
+    fn default() -> Self {
+        let tokio_runtime =
+            tokio::runtime::Runtime::new().expect("Could not create internal Tokio runtime");
+
+        let docker = match env::var("DOCKER_HOST") {
+            #[cfg(target_family = "unix")]
+            Ok(docker_host) if docker_host.starts_with("unix://") => {
+                Docker::connect_with_unix_defaults()
+            }
+            Ok(docker_host)
+            if docker_host.starts_with("tcp://") || docker_host.starts_with("https://") =>
+                {
+                    #[cfg(not(feature = "remote-docker"))]
+                    panic!("Cannot connect to DOCKER_HOST '{docker_host}' since it requires TLS. Please use a local Docker daemon instead (recommended), or else enable the experimental `remote-docker` feature.");
+                    #[cfg(feature = "remote-docker")]
+                    Docker::connect_with_ssl_defaults()
+                }
+            Ok(docker_host) => panic!("Cannot connect to unsupported DOCKER_HOST '{docker_host}'"),
+            Err(VarError::NotPresent) => Docker::connect_with_local_defaults(),
+            Err(VarError::NotUnicode(_)) => {
+                panic!("DOCKER_HOST environment variable is not unicode encoded!")
+            }
+        }
+            .expect("Could not connect to local Docker daemon");
+
+        TestRunner::new(tokio_runtime, docker)
+    }
+}
+
+impl TestRunner {
+    /// Creates a new runner that uses the given Tokio runtime and Docker connection.
+    ///
+    /// This function is meant for advanced use-cases where fine control over the Tokio runtime
+    /// and/or Docker connection is required. For the common use-cases, use `Runner::default`.
+    pub fn new(tokio_runtime: tokio::runtime::Runtime, docker: Docker) -> Self {
```

Comment:
I like the idea of a default TestRunner and the ability to create your own if you ever need to. Since this isn't a huge maintenance burden (we simply expose something we need in any case) I'll keep it.